### PR TITLE
Adding service definition and SLO definition

### PR DESCRIPTION
**Pull request by yuriatgoogle**
Fixes #

## Proposed Changes

* added service definition in slos.tf
* added availability SLO in slos.tf
*

**Release Note**

```release-note
Added service definition and availability SLO definition
```

### CODE REVIEW

File: terraform/alerting/verification-server/slos.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  # the basics
+  service = google_monitoring_custom_service.verification-server.service_id
+  slo_id = "availability-slo"
+  display_name = "99.9% of requests are successful over rolling 28 days"
+  project = var.monitoring-host-project
+
+  # the SLI
+  request_based_sli {
+    good_total_ratio {
+      good_service_filter = join(" AND ", [
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:31]: Use multiline string literal, there's no need to join here:

good_service_filter = <--EOT
metric.type="loadbalancing.googleapis.com/https/request_count" AND
...
EOT

File: terraform/alerting/verification-server/slos.tf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "metric.type=\"loadbalancing.googleapis.com/https/request_count\"",
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:32]: It looks like this depends on GCLB? If so please optionally enable it based on whether var.https-forwarding-rule is defined or not. See `alerts.tf` for reference.

File: terraform/alerting/verification-server/alerts.tf [hunk 3]
```diff
@@ -326,4 +326,4 @@ EOT
     null_resource.manual-step-to-enable-workspace,
     google_logging_metric.stackdriver_export_error_count
   ]
-}
+}
```
Comment (yegle) [inline terraform/alerting/verification-server/alerts.tf:329]: Revert this change.

File: terraform/alerting/verification-server/slos.tf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_monitoring_custom_service" "verification-server" {
+  service_id = "verification-server"
+  display_name = "Verification Server"
+  project = var.monitoring-host-project
+}
+
+resource "google_monitoring_slo" "availability-slo" {
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:21]: Will there be a notification if we are out of SLO?

Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf:21]: Yes, I'll add an alert in a subsequent PR.

File: terraform/alerting/verification-server/slos.tf (hunk 1 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf:31]: Done.

File: terraform/alerting/verification-server/slos.tf (hunk 2 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf]: Done - I think?

File: terraform/alerting/verification-server/alerts.tf (hunk 3 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/alerts.tf]: Done, I think?

**Comment by yuriatgoogle** [comment]
I've tried to address all comments.

File: terraform/alerting/verification-server/slos.tf [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  # the basics
+  service = google_monitoring_custom_service.verification-server.service_id
+  slo_id = "availability-slo"
+  display_name = "99.9% of requests are successful over rolling 28 days"
+  project = var.monitoring-host-project
+  # create if GCLB is used
+  depends_on = [
+    var.https-forwarding-rule
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:29]: Use count to optionally enable this resource:

```
count = var.https-forwarding-rule == "" ? 0 : 1
```

File: terraform/alerting/verification-server/slos.tf [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        metric.type="loadbalancing.googleapis.com/https/request_count"
+        resource.type="https_lb_rule"
+        resource.label.backend_name="apiserver"
+        metric.label.response_code_class=200
+      EOT
+      bad_service_filter = <<-EOT
+        metric.type="loadbalancing.googleapis.com/https/request_count"
+        resource.type="https_lb_rule"
+        resource.label.backend_name="apiserver"
+        metric.label.response_code_class=500
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:45]: What if we accidentally rejected all requests with 429 status code?

I think there's some nuance here if we only look at 200 and 500 code. I'm fine with only check 200/500 but am a bit concern that it doesn't give us the whole picture.

File: terraform/alerting/verification-server/slos.tf [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      EOT
+    }
+  }
+
+  # the goal
+  goal = 0.999
```
Comment (yegle) [inline terraform/alerting/verification-server/slos.tf:51]: Let's start with 0.995 first. Most of our dependencies have SLA of 0.995 (Cloud Run, GCLB etc.)

**Review by yegle — APPROVED** [review]

**Comment by yegle** [comment]
Oops I didn't realize approval will auto merge. Please address the depends_on in a follow up PR.

File: terraform/alerting/verification-server/slos.tf (hunk 5 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf:29]: Done - I think?

File: terraform/alerting/verification-server/slos.tf (hunk 7 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf:51]: Done.

File: terraform/alerting/verification-server/slos.tf (hunk 6 above)
Comment (yuriatgoogle) [reply terraform/alerting/verification-server/slos.tf:45]: I agree that this is an imperfect place to start. Let's see what alerts look like once this is in place.
